__tests__/app.test.js

```diff
@@ -308,3 +308,111 @@ describe("GET /api/users should return a list of all users", () => {
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("votes", { descending: true });
+				}),
+			request(app)
+				.get("/api/reviews?sortBy=review_id")
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("review_id", { descending: true });
+				}),
+			request(app)
+				.get("/api/reviews?sortBy=category")
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("category", { descending: true });
+				}),
+		]);
+	});
+	test("200 if given the correct orderBy values", () => {
+		return Promise.all([
+			request(app)
+				.get("/api/reviews?sortBy=votes&orderBy=ASC")
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("votes", { descending: false });
+				}),
+			request(app)
+				.get("/api/reviews?sortBy=review_id&orderBy=DESC")
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("review_id", { descending: true });
+				}),
+			request(app)
+				.get("/api/reviews?sortBy=category&orderBy=ASC")
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("category", { descending: false });
+				}),
+		]);
+	});
+	test("200 if given the correct category values", () => {
+		return Promise.all([
+			request(app)
+				.get("/api/reviews?sortBy=votes&category='social deduction'")
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("votes", { descending: true });
+					body.reviews.forEach((game) => {
+						expect(game.category).toEqual("social deduction");
+					});
+				}),
+			request(app)
+				.get("/api/reviews?sortBy=votes&orderBy=ASC&category='dexterity'")
+
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("votes");
+					body.reviews.forEach((game) => {
+						expect(game.category).toEqual("dexterity");
+					});
+				}),
+			request(app)
+				.get("/api/reviews?sortBy=category&orderBy=ASC")
+				.expect(200)
+				.then(({ body }) => {
+					expect(body.reviews).toBeSortedBy("category", { descending: false });
+				}),
+		]);
+	});
+	test("400 if sortBy, orderBy or category is an invalid datatype", () => {
+		return Promise.all([
+			request(app)
+				.get("/api/reviews?sortBy=bestdog")
+				.expect(400)
+				.then(({ body }) => {
+					expect(body).toEqual({ msg: "Bad request" });
+				}),
+			request(app)
+				.get("/api/reviews?orderBy=birthday")
+				.expect(400)
+				.then(({ body }) => {
+					expect(body).toEqual({ msg: "Bad request" });
+				}),
+		]);
+	});
+	test("404 if sortBy or orderBy is invalid", () => {
+		return Promise.all([
+			request(app)
+				.get("/api/reviews?category='birthday'")
+				.expect(404)
+				.then(({ body }) => {
+					expect(body).toEqual({ msg: "Not found" });
+				}),
+			request(app)
+				.get("/api/reviews?category=7")
+				.expect(404)
+				.then(({ body }) => {
+					expect(body).toEqual({ msg: "Not found" });
+				}),
```

Comment:
Great testing so far James! Could I just request one more please...

- Status 200, valid category query, but has no reviews responds with an empty array of articles, e.g. ?category=children's games.